src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/Instrument.common.cs

```diff
@@ -52,6 +52,11 @@ protected Instrument(Meter meter, string name, string? unit, string? description
         /// <param name="tags">A span of key-value pair tags associated with the measurement.</param>
         protected void RecordMeasurement(T measurement, ReadOnlySpan<KeyValuePair<string, object?>> tags)
         {
+            if (!Meter.IsSupported)
```

Comment:
Can we avoid adding this check and count on _subscription.First to be null? It looks like you already blocked all the code paths that could have registered a subscription.

Reply:
Yes, we can do that. According to my understanding (per @eerhardt explanation) jit is going to optimize that in tier two and will fully remove this check. I was hoping when `IsSupported` is false, the linker can empty the whole method body including accessing the static fields. Let me know if you still think we should remove this check.

Reply:
Can you confirm the JIT does optimize it as expected. As long as that is confirmed then I have no qualms leaving it there :)

Reply:
Adding this check will allow the trimmer to remove the rest of the method body, right?

Reply:
> Adding this check will allow the trimmer to remove the rest of the method body, right?

Correct - and all the methods it calls that aren't called elsewhere.

Reply:
I created a simple app using my private built. The app and the runtime were built as a release. I have `IsSupported = true` to test the normal case. I run it under the debugger, and I am seeing the call `Meter.IsSupported` is not removed inside `RecordMeasurement` method. I'll remove this check from there as this is a hot path, we want to avoid regressions there. 

![image](https://github.com/dotnet/runtime/assets/10833894/85bd2d02-0903-47f8-ba83-525c8d948f7f)